src/nvim/fileio.c

```diff
@@ -503,13 +499,10 @@ readfile (
     fd = os_open((char *)fname, O_RDONLY, 0);
   }
 
-  if (fd < 0) {                     /* cannot open at all */
+  if (fd < 0) {                     // cannot open at all
     msg_scroll = msg_save;
 #ifndef UNIX
-    /*
-     * On non-unix systems we can't open a directory, check here.
-     */
-    perm = os_getperm(fname);      /* check if the file exists */
```

Comment:
This isn't needed anymore because:

- on line 420 (the `(!read_buffer && !read_stdin)` block) we set it for all platforms
- `fd` will be `0` unless line 499 is reached--which is _also_ guarded by `(!read_buffer && !read_stdin)` condition